src/Simulations/run.jl

```diff
@@ -103,7 +103,14 @@ function run!(sim; pickup=false)
 
     if we_want_to_pickup(pickup)
         checkpointers = filter(writer -> writer isa Checkpointer, collect(values(sim.output_writers)))
-        set!(model, checkpoint_path(pickup, checkpointers))
+        checkpoint_filepath = checkpoint_path(pickup, checkpointers)
+
+        # https://github.com/CliMA/Oceananigans.jl/issues/1159
+        if pickup isa Bool && isnothing(checkpoint_filepath)
+            @warn "pickup=true but no checkpoints ware found. Simulation will run without picking up."
+        else
+            set!(model, checkpoint_filepath)
```

Comment:
Does this work if `checkpoint_filepath=nothing` but `pickup` is not a boolean?

Reply:
It would but I decided to only allow this behavior when `pickup=true`, but maybe it's more consistent if we allow it for iteration numbers and filepaths as well? Although it could seem weird to keep running when you wanted to pickup from a specific checkpoint.

Reply:
Just trying to understand the flow of information. Reading this code it seems possible that pickup might be something other than a bool when `checkpoint_filepath===nothing`. But looking deeper it seems this is not the case it looks like `checkpoint_filepath` can only be `nothing` in the case that `pickup isa Bool`. So I don't know if I'm missing something.